x-pack/plugins/infra/public/utils/logs_overview_fetchers.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraClientCoreSetup } from '../types';
+
+// TODO: this satisfies a TS error because TS inference assumed the stat type was just "string",
+// but we need a better solution for this 🤔
+type StatType = 'number' | 'percent' | 'bytesPerSecond';
```

Comment:
You can solve it by specifying the return type of `getLogsOverviewDataFetcher`.
```suggestion
export function getLogsOverviewDataFetcher(
  getStartServices: InfraClientCoreSetup['getStartServices']
): Promise<LogsFetchDataResponse>
```